### ISSUE TITLE
sublists

### ISSUE THREAD

**Issue by KonoromiHimaries** [yourduskquibbles/webannoyances#67]
do you can divide the webannoyances list into a sublists

example. you can based on the model of polishsubfilters
https://github.com/KonoromiHimaries/PolishSubFilters/tree/master/internal-filters
or polishannoyancefilters model
https://github.com/PolishFiltersTeam/PolishAnnoyanceFilters/tree/master/PPB

potential benefits
- fast loading of the list
- increased editing convenience
- anyone will be able to subscribe to one sublist without having to subscribe to all webannoyances filters

**Comment by yourduskquibbles** [comment]
I would ultimately like to do this but need to research how some of the other lists maintainers manage this before doing so. 

It's also a time commitment that I don't have right now to implement immediately.

**Comment by hawkeye116477** [comment]
You can just divide it into few files, then use `#include` directives in big main filterlist, which uBO supports to include smaller lists. Of course you can also do that just like I created templates file, made some scripts to merge all, but I think that in that case it's not needed, cuz you support only uBO, but maybe autosorting and autoversioning will be useful :smile:

**Comment by yourduskquibbles** [comment]
Forgot this was still open, thanks for closing.

**Comment by yourduskquibbles** [comment]
Moving forward, I will be moving empty advertising filters into the [`ad_placeholders.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/ad_placeholders.txt) sublist.

The long-term intention with [`ad_placeholders.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/ad_placeholders.txt) sublist will be to remove all of these advertising filters from [`utlralist.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt) so eventually [`ultralist.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt) is ONLY a list that exists to subscribes to all [Web Annoyances sublists](https://github.com/yourduskquibbles/webannoyances/tree/master/filters). 

This should allow the flexibility for users to only subscribe to certain features of [Web Annoyances Ultralist](https://github.com/yourduskquibbles/webannoyances) e.g. users who only want to remove social buttons & empty ad boxes but don't want to change any CSS elements of a site like sticky headers or newsletter subscription requests (e.g. don't subscribe to [`css_style_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/css_style_filters.txt) & [`newsletter_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/newsletter_filters.txt))

This is also the reason behind recently creating [`css_companion_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/css_companion_filters.txt) sublist.

**Comment by yourduskquibbles** [comment]
Reopening to help others track maintenance progress of moving existing filters from [`ultralist.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt) into various sublists.

**Comment by GeneralKenobi1** [comment]
It might make sense to give cookie/gdpr filters their own sublist.

Also, I see a bunch of social filters in the generic sublist.

I'd be happy to submit a pull request for one or both of these if you're interested.

**Comment by llacb47** [comment]
Should paywall filters go in a separate list too? And what's the policy in this repo? Are all paywalls removed when possible?

**Comment by yourduskquibbles** [comment]
@GeneralKenobi1 I agree with splitting up cookie/gdpr filters - I can create a new sublist for that and I would be happy to accept a PR - Right now most of those filters would be in the [`modal_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/modal_filters.txt) list as well as [`generic_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/generic_filters.txt)

@llacb47 I prefer not to have a separate paywall sublist, the general policy for Web Annoyances Ultralist is to remove them if it's a soft paywall/warning that only requires modifying CSS or blocking a 1st or 3rd party script - no script injection workarounds or faking subscription status. 

There are separate extensions users can use if they want to go that route, e.g. [Bypass Paywalls Clean for Firefox](https://bitbucket.org/magnolia1234/bypass-paywalls-firefox-clean/src/master/README.md) & [Bypass Paywalls Clean for Chrome](https://bitbucket.org/magnolia1234/bypass-paywalls-chrome-clean/src/master/README.md) or even something as simple as [Cookie Auto Delete for Firefox](https://addons.mozilla.org/en-US/firefox/addon/cookie-autodelete/) & [Cookie Auto Delete for Chrome](https://chrome.google.com/webstore/detail/cookie-autodelete/fhcgjolkccmbidfldomjliifgaodjagh/)

**Comment by llacb47** [comment]
There's a scriptlet bypass right here, are you going to remove it? :P https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt#L185

**Comment by yourduskquibbles** [comment]
@GeneralKenobi1 I just [added](https://github.com/yourduskquibbles/webannoyances/commit/8a0a299bab3ea67a533506811d42b938575807c1) a new [`cookie_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/cookie_filters.txt) sublist and just seeded it with a few existing filters from the [`modal_filters.txt`](https://github.com/yourduskquibbles/webannoyances/blob/master/filters/modal_filters.txt) sublist. Feel free to submit PR's and THANK YOU in advance for any help!

**Comment by yourduskquibbles** [comment]
> There's a paywall scriptlet bypass right here, are you going to remove it? :P https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt#L185

Looks like that is just removing a cosmetic DOM element, not injecting scripts or changing the page content that isn't already loaded in the DOM when a user navigates to the page?

**Comment by llacb47** [comment]
> > There's a paywall scriptlet bypass right here, are you going to remove it? :P https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt#L185
> 
> Looks like that is just removing a cosmetic DOM element, not injecting scripts or changing the page content that isn't already loaded in the DOM when a user navigates to the page?

https://github.com/yourduskquibbles/webannoyances/blob/8a0a299bab3ea67a533506811d42b938575807c1/ultralist.txt#L186

**Comment by yourduskquibbles** [comment]
> 
> 
> > > There's a paywall scriptlet bypass right here, are you going to remove it? :P https://github.com/yourduskquibbles/webannoyances/blob/master/ultralist.txt#L185
> > 
> > 
> > Looks like that is just removing a cosmetic DOM element, not injecting scripts or changing the page content that isn't already loaded in the DOM when a user navigates to the page?
> 
> https://github.com/yourduskquibbles/webannoyances/blob/8a0a299bab3ea67a533506811d42b938575807c1/ultralist.txt#L186

I see you pointed to L186 in edit now, yes I will remove that, no need to create undue risk when other extensions exist. Probably added it a long time ago when I found the filter on reddit or something like that. Thanks for pointing out

**Comment by yourduskquibbles** [comment]
> I'm not saying you should remove it by the way, just saying this policy limits how much paywall bypassing can be done

It's OK, it's not the intention or primary goal of this list to bypass paywalls so don't want to host that anyway and risk taking the whole project down.

**Comment by ghost** [comment]
> You can just divide it into few files, then use `#include` directives in big main filterlist, which uBO supports to include smaller lists. Of course you can also do that just like I created templates file, made some scripts to merge all, but I think that in that case it's not needed, cuz you support only uBO, but maybe autosorting and autoversioning will be useful 😄

@hawkeye116477 What exactly do you mean by autosorting?

**Comment by hawkeye116477** [comment]
> @hawkeye116477 What exactly do you mean by autosorting?

I mean that script will sort rules alphabetically. You can then set launching it through GitHub Actions for example twice per day, so it will be automatic process.
